catalyst_voices/packages/internal/catalyst_voices_view_models/lib/src/proposal/proposal_view_model.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_view_models/src/campaign/campaign_stage.dart';
+import 'package:catalyst_voices_view_models/src/proposal/funded_proposal.dart';
+import 'package:catalyst_voices_view_models/src/proposal/pending_proposal.dart';
+import 'package:equatable/equatable.dart';
+
+/// A proposal view model spanning proposals in different stages.
+sealed class ProposalViewModel extends Equatable {
```

Comment:
I'm not 100% sure if we need this class. It does nothing but wraps `PendingProposal` and `FundedProposal`. If so then simple interface / base class would be sufficient here.

smth like this

```dart
abstract base class ProposalViewModel extends Equatable {
  final String id;

  const ProposalViewModel({
    required this.id,
  });
}

final class FundedProposal extends ProposalViewModel {
  const FundedProposal({
    required super.id,
}

final class PendingProposal extends ProposalViewModel {
  const PendingProposal({
    required super.id,
}
```